shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/Linker.kt

```diff
@@ -186,11 +195,18 @@ open class MacOSBasedLinker(targetProperties: AppleConfigurables)
 
     override fun filterStaticLibraries(binaries: List<String>) = binaries.filter { it.isUnixStaticLib }
 
-    override fun linkCommands(objectFiles: List<ObjectFile>, executable: ExecutableFile,
-                              libraries: List<String>, linkerArgs: List<String>,
-                              optimize: Boolean, debug: Boolean, kind: LinkerOutputKind,
-                              outputDsymBundle: String,
-                              needsProfileLibrary: Boolean): List<Command> {
+    override fun preLinkCommands(objectFiles: List<ObjectFile>, output: ObjectFile): List<Command> =
+        Command(linker).apply {
```

Comment:
I've encountered some issues when not specifying some of the common flags like -arch for ld -r. We don't have caches for non-x86_64-arches though.

Have you checked ld flags when Xcode is prelinking?